unit_test/integration/test_heat_demand.py

```diff
@@ -0,0 +1,144 @@
+import pandas as pd
+
+from typing import Tuple
+
+from omotes_simulator_core.adapter.transforms.mappers import (
+    EsdlControllerMapper
+)
+from omotes_simulator_core.entities.esdl_object import EsdlObject
+from omotes_simulator_core.entities.simulation_configuration import SimulationConfiguration
+from omotes_simulator_core.infrastructure.simulation_manager import SimulationManager
+from omotes_simulator_core.infrastructure.utils import pyesdl_from_file
+from omotes_simulator_core.solver.utils.fluid_properties import fluid_props
+
+
+class HeatDemandTest(unittest.TestCase):
+    """Class to store the heat demand tests."""
+
+    def setUp(self) -> None:
+        """Setting up and running the simulation used for the tests."""
+        esdl_file_path = str(Path(__file__).parent / ".." / ".." / "testdata" / "test1.esdl")
+        esdl_object = EsdlObject(pyesdl_from_file(esdl_file_path))
+        self.controller = EsdlControllerMapper().to_entity(esdl_object)
+        self.start_time = datetime.strptime("2019-01-01T00:00:00", "%Y-%m-%dT%H:%M:%S")
+        self.end_time = datetime.strptime("2019-01-01T06:00:00", "%Y-%m-%dT%H:%M:%S")
+        config = SimulationConfiguration(
+            simulation_id=uuid.uuid1(),
+            name="test run",
+            timestep=3600,
+            start=self.start_time,
+            stop=self.end_time
+        )
+        callback = Mock()
+        app = SimulationManager(EsdlObject(pyesdl_from_file(esdl_file_path)), config)
+        self.result = app.execute(callback)
+
+        energy_system = esdl_object.energy_system_handler.energy_system
+
+        # Collect demands
+        self.demands = []
+        for element in energy_system.eAllContents():
+            if isinstance(element, esdl.HeatingDemand):
+                self.demands.append(element)
+
+        # Collect id of in and out ports of each demand.
+        self.in_out_demand_dict = dict()
+        for demand in self.demands:
+            self.in_out_demand_dict[demand.id] = dict()
+            for port in demand.port:
+                if isinstance(port, esdl.InPort):
+                    self.in_out_demand_dict[demand.id]["InPort"] = port.id
+                elif isinstance(port, esdl.OutPort):
+                    self.in_out_demand_dict[demand.id]["OutPort"] = port.id
+
+    def _get_demand_in_out_temperatures(self, in_out_ports: dict) -> Tuple[dict, dict]:
+        """Gets the temperatures at the in and out ports of the demand assets in the simulation."""
+        temp_in_dict = dict()
+        temp_out_dict = dict()
+        for demand in self.demands:
+            in_port_id = in_out_ports[demand.id]["InPort"]
+            out_port_id = in_out_ports[demand.id]["OutPort"]
+
+            temp_in_dict[demand.id] = np.array(self.result[(in_port_id, 'temperature')])
+            temp_out_dict[demand.id] = np.array(self.result[(out_port_id, 'temperature')])
+
+        return temp_in_dict, temp_out_dict
+
+    def test_heat_power(self) -> None:
```

Comment:
I do not get why you want to test this, since it might be that the production is to low so the consumption is capped, and this test will fail.